Tests/BugsnagMetadataTest.m

```diff
@@ -0,0 +1,184 @@
+//
+//  BugsnagMetadataTest.m
+//  Tests
+//
+//  Created by Robin Macharg on 28/01/2020.
+//  Copyright © 2020 Bugsnag. All rights reserved.
+//
+
+#import "BugsnagMetadata.h"
+#import <XCTest/XCTest.h>
+
+// MARK: - Expose tested-class internals
+
+@interface BugsnagMetadataTest : XCTestCase <BugsnagMetadataDelegate>
+@property BOOL delegateCalled;
+@end
+
+@interface BugsnagMetadata ()
+@property(atomic, strong) NSMutableDictionary *dictionary;
+@end
+
+// MARK: - DummyClass
+
+@interface DummyClass : NSObject <NSCopying> // <NSCopying> allows it to be used as a dictionary key
+@property NSString *name;
+@end
+
+@implementation DummyClass
+@synthesize name;
+- (nonnull id)copyWithZone:(nullable NSZone *)zone {
+    DummyClass *copy = [DummyClass new];
+    copy.name = self.name;
+    return copy;
+}
+
+@end
+
+// MARK: - Tests
+
+@implementation BugsnagMetadataTest
+
+@synthesize delegateCalled;
+
+- (void)test_addAttribute_withName {
```

Comment:
The contents of this test looks good - if it's straightforward would it be possible to separate these test cases into individual functions? E.g., one for creation, one for checking an arbitrary tab name creates itself, etc.

Reply:
NP, extended setUp() slightly, split tests up and adjusted any cumulative values being tested to account for this.